pykechain/client.py

```diff
@@ -2083,6 +2085,13 @@ def create_property(
                 )
             options.update(scope_options)
 
+        elif property_type == PropertyType.DATETIME_VALUE:
+            default_value = check_datetime(dt=default_value, key="default_value")
+        elif property_type == PropertyType.DATE_VALUE:
+            default_value = check_date(dt=default_value, key="default_value")
+        elif property_type == PropertyType.TIME_VALUE:
+            default_value = check_time(dt=default_value, key="default_value")
```

Comment:
I'm not that well acquainted with pykechain code so i can't say it's wrong but seeing such `if/else` blocks makes me suspicious.

![suspicious](https://media.giphy.com/media/ANbD1CCdA3iI8/giphy.gif)

Reply:
I thought about using the new pattern matching released in python 3.10: https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching

But that would break for all the other python version which we are still supporting in the scripts. Why does it make you suspicious otherwise?

Reply:
Such `if/else` blocks usually are signs for violation of design principles.
In this concrete case i _think_ the whole `if/else` is just responsible for validating/cleaning the data.
So different `PropertyType` should know how to validate input data without conditionals.
Thats something to keep in mind for later. It's not a blocking issue.